app/agents/JobOfferManagerAgent.py

```diff
@@ -192,4 +214,118 @@ async def run(self):
 
-        self.agent.applications_to_analyze = None
\ No newline at end of file
+        self.agent.applications_to_analyze = None
+
+
+class GetStatusResponse(spade.behaviour.CyclicBehaviour):
+    """
+    Sends analyze request to RecruiterManager
+    Protocols/Activities in GAIA (role JobOfferManager): GetStatusRequest, GetStatusResponse
+    """
+
+    agent: JobOfferManagerAgent
+
+    async def run(self):
+        msg = await self.receive(timeout=120)
+        if msg is None:
+            return
+
+        self.agent.logger.info("Received get status request from %s.", msg.sender)
+        type, _  = await self.agent.get_message_type_and_data(msg) 
+
+        if type != MessageType.STATUS_REQUEST:
+            self.agent.logger.warning("Received an unknown or invalid message.")
+            return
+        
+        jobOffer = self.agent.jobOffer
+        if jobOffer is None:
+            self.agent.logger.error("Job offer not found.")
+            await self.agent.stop()
+
+        finished_applications = [
+            x for x in jobOffer.applications if x.status == ApplicationStatus.FINISHED
+        ]
+        analysed_applications = [
+            x for x in jobOffer.applications if x.status == ApplicationStatus.ANALYZED
+        ]
+        rejected_applications = [
+            x for x in jobOffer.applications if x.status == ApplicationStatus.REJECTED
+        ]
+        
+        msg = await self.agent.prepare_message(
+            f"{msg.sender}", 
+            "response", 
+            "status", 
+            MessageType.STATUS_RESPONSE,
+            [f"{jobOffer._id}",f"{jobOffer.name}", f"{jobOffer.status.value}", f"{jobOffer.description}", 
+             f"{len(finished_applications)}",f"{len(analysed_applications)}", f"{len(rejected_applications)}",
+             f"{jobOffer.best_candidate_id}" ]
+        )
+
+        await self.send(msg)
+
+        self.agent.logger.info("Sent message status result to %s.", msg.sender)
+
+
+class FinishedRecruitmentRequest(spade.behaviour.PeriodicBehaviour):
+    """
+    Sends analyze request to RecruiterManager
+    Protocols/Activities in GAIA (role JobOfferManager): FinishedRecruitmentRequest, UpdateCandidatesRanking, CandidatesResult, 
+    ChooseBestCandidates, InformRecruiter, QueueResultNotification, Close
+    """
+
+    agent: JobOfferManagerAgent
+
+    async def run(self):
+        finished_applications = self.agent.jobOfferModule.get_finished_applications(self.agent.job_offer_id)
+        
+        if(len(finished_applications)== self.agent.jobOffer.max_number_of_candidates and self.agent.jobOffer.status != JobOfferStatus.CLOSED):
+            self.agent.logger.info("Closing job offert")
+            self.agent.jobOfferModule.change_job_offer_status(self.agent.job_offer_id, JobOfferStatus.CLOSED)  
+            await self.chooseBestCandidate(finished_applications)
+            await self.sendNotifToAllCandidates()
```

Comment:
po to byl ten status `TOCLOSE` zeby wlasnie tu ustawic a `CLOSED` dopiero po wybraniu besta, ale to w sumie drobiazg, mi się akurat trafił raport w logach Rera że offer jest closed i nie ma best candidata

Reply:
Po rozmowie na ten moment zamykam